### ISSUE TITLE
Min/Max System

### ISSUE THREAD

**Issue by jaghachi** [inventree/InvenTree#1150]
Hello, I noticed you only have a variable for "minimum_stock: Minimum preferred quantity to keep in stock". 

May I make a suggestion about have max variable as well?

Thank you for this wonderful project!

Sorry @SchrodingersGat I wasn't sure where else to suggest this. Github has "discussions" in beta and I can maybe get them to enable it for this project if you'd like.

**Comment by SchrodingersGat** [comment]
This is the ideal place to suggest this 👍 

Can you explain your use-case for having a maximum stock setting?

**Comment by jaghachi** [comment]
Great! Sure so I'm setting up a warehouse with limited stocking space. For example we have allocated 100 paint cans of a certain color to a space. I need to make sure I don't over stock.

**Comment by SchrodingersGat** [comment]
I see, this makes sense!

**Comment by SchrodingersGat** [comment]
@jaghachi still interested in this one? It has languished for a while.

**Comment by Spritz2k20** [comment]
> [@jaghachi](https://github.com/jaghachi) still interested in this one? It has languished for a while.

I'm thinking about switching our current inventory management system over to InvenTree. But I'm honestly surprised that this max function doesn’t exist. In our current system, every product has a minimum and a maximum stock level. When the stock drops below the minimum, it automatically orders more to bring it back up to the maximum.

We manage around 1,000 different products, so it’s essential that this happens automatically.

It would be great if you could implement this!

This is by the way used in a feature that generates suggested purchase orders for each supplier who sells one or more of the products that are below their minimum stock level.

**Comment by SchrodingersGat** [comment]
@Spritz2k20 what is the use case for the "high stock" level in your application?

**Comment by Spritz2k20** [comment]
> [@Spritz2k20](https://github.com/Spritz2k20) what is the use case for the "high stock" level in your application?

Thanks for your quick reply!

We use minimum and maximum (desired/target) stock levels. The system monitors the stock levels of all products, and when the stock drops below the defined minimum stock, it automatically creates a suggested purchase order for each supplier who provides products that need replenishment.

The idea is to order not just enough to reach the minimum level, but to bring the stock back up to the maximum (or target) level. This helps reduce how often we need to place orders.

By setting the maximum stock level appropriately, we can control how often we need to order. For example, if we set it right, we can make sure that most products only need to be ordered once a month.

In our current system, these suggested purchase orders can then easily be converted into actual purchase orders.

**Comment by SchrodingersGat** [comment]
Makes sense, thanks. Having an "excess stock" callout for those parts with defined "max" levels would be good.

Would you be interested in implementing this? It should be a straightforward feature to implement and I'm happy to provide guidance.

**Comment by Spritz2k20** [comment]
There's a good chance I will!

But I’ll need to find the time for it, and there are quite a few projects I need to wrap up before I can get to this.

Hopefully, I’ll be able to get back to you at some point in the future.

**Comment by SchrodingersGat** [comment]
Hi @Spritz2k20 just checking in to see if you are still interested in developing this?

**Comment by Spritz2k20** [comment]
> this

Yes, probably. But not in the near future. It got bumped way down on my to do list.

**Comment by matmair** [comment]
This was implemented an will be in the 1.4.0 release
